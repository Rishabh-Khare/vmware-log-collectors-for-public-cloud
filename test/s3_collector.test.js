/*
Copyright 2018 VMware, Inc.
SPDX-License-Identifier: MIT
*/

const { lintTestEnv } = require('../test/helper.test');
const { s3SampleData } = require('./sample_files/s3_testdata');
const { s3SampleFileContent } = require('./sample_files/s3_testdata');
const { s3SampleRawLogs } = require('./sample_files/s3_testdata');

const {
  S3HttpCollector,
} = require('../src/index');

describe('S3HttpCollector', () => {
  const tagRegexMap = new Map(Object.entries({
    tag1: new RegExp('c(k+)c', 'i'),
  }));
  const collector = new S3HttpCollector(lintTestEnv, tagRegexMap);

  describe('processLogsJson', () => {
    it('should process logs JSON', () => {
      const logsJson = s3SampleData();
      collector.processLogsJson(logsJson);
      expect(logsJson)
        .toMatchSnapshot();
    });
  });

  describe('processS3Data', () => {
    it('should process JSON data', () => {
      const logsJson = s3SampleFileContent();
      const data = collector.processS3Data('test', JSON.stringify(logsJson));
      expect(data)
        .toMatchSnapshot();
    });

    it('should process PlainText data', () => {
      const rawLogs = s3SampleRawLogs();
      const data = collector.processS3Data('test', JSON.stringify(rawLogs));
      expect(data)
        .toMatchSnapshot();
    });
  });
});
