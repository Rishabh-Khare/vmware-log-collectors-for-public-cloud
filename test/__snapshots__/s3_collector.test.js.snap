// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`S3HttpCollector processLogsJson should process logs JSON 1`] = `
Object {
  "Records": Array [
    Object {
      "awsRegion": "us-west-2",
      "bucket_arn": "arn:aws:s3:::sourcebucket",
      "bucket_name": "sourcebucket",
      "bucket_ownerIdentity_principalId": "A3NL1KOZZKExample",
      "configurationId": "testConfigRule",
      "eventName": "ObjectCreated:Put",
      "eventSource": "aws:s3",
      "eventTime": "1970-01-01T00:00:00.000Z",
      "eventVersion": "2.0",
      "ingest_timestamp": 1538769915817,
      "log_type": "aws_s3",
      "object_eTag": "d41d8cd98f00b204e9800998ecf8427e",
      "object_key": "HappyFace.jpg",
      "object_size": 1024,
      "object_versionId": "096fKKXTRTtl3on89fVO.nfljtsv6qko",
      "requestParameters_SourceIPAddress": "127.0.0.1",
      "responseElements_Xamzid2": "FMyUVURIY8/IgAtTv8xRjskZQpcIZ9KG4V5Wp6S7S/JRWeUWerMUE5JgHvANOjpD",
      "responseElements_Xamzrequestid": "C3D13FE58DE4C810",
      "s3SchemaVersion": "1.0",
      "text": Object {
        "bucket": Object {
          "arn": "arn:aws:s3:::sourcebucket",
          "name": "sourcebucket",
          "ownerIdentity": Object {
            "principalId": "A3NL1KOZZKExample",
          },
        },
        "configurationId": "testConfigRule",
        "object": Object {
          "eTag": "d41d8cd98f00b204e9800998ecf8427e",
          "key": "HappyFace.jpg",
          "size": 1024,
          "versionId": "096fKKXTRTtl3on89fVO.nfljtsv6qko",
        },
        "s3SchemaVersion": "1.0",
      },
      "userIdentityPrincipalId": "AIDAJDPLRKLG7UEXAMPLE",
    },
  ],
}
`;

exports[`S3HttpCollector processS3Data should process JSON data 1`] = `
Array [
  Object {
    "bucket_name": "test",
    "ingest_timestamp": 1538769915817,
    "log_type": "aws_s3data",
    "text": Object {
      "0_agent": "Debian APT-HTTP/1.3 (0.8.16~exp12ubuntu10.21)",
      "0_bytes": 0,
      "0_referrer": "-",
      "0_remote_ip": "93.180.71.3",
      "0_remote_user": "-",
      "0_request": "GET /downloads/product_1 HTTP/1.1",
      "0_response": 304,
      "0_time": "17/May/2015:08:05:32 +0000",
      "1_agent": "Debian APT-HTTP/1.3 (0.8.16~exp12ubuntu10.21)",
      "1_bytes": 0,
      "1_referrer": "-",
      "1_remote_ip": "93.180.71.3",
      "1_remote_user": "-",
      "1_request": "GET /downloads/product_1 HTTP/1.1",
      "1_response": 304,
      "1_time": "17/May/2015:08:05:23 +0000",
    },
  },
]
`;

exports[`S3HttpCollector processS3Data should process PlainText data 1`] = `
Array [
  Object {
    "bucket_name": "test",
    "ingest_timestamp": 1538769915817,
    "log_type": "aws_s3data",
    "text": Object {
      "0": "83.149.9.216 - - [17/May/2015:10:05:03 +0000] \\"GET /presentations/logstash-monitorama-2013/images/kibana-search.png HTTP/1.1\\" 200 203023 \\"http://semicomplete.com/presentations/logstash-monitorama-2013/\\" \\"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_9_1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/32.0.1700.77 Safari/537.36\\"",
      "1": "83.149.9.216 - - [17/May/2015:10:05:43 +0000] \\"GET /presentations/logstash-monitorama-2013/images/kibana-dashboard3.png HTTP/1.1\\" 200 171717 \\"http://semicomplete.com/presentations/logstash-monitorama-2013/\\" \\"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_9_1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/32.0.1700.77 Safari/537.36\\"",
    },
  },
]
`;
